### DISCUSSION TITLE
I got a problem when i try to deploying a simple Solidity contract to zkSync Testnet using the zkSync Remix plugin

### DISCUSSION THREAD

**Discussion by seni7** [Cyfrin/foundry-full-course-cu#3030] [Q&A] [answered]
it shows 
Expectation Failed
Solidity Verification Failed, logs can be read in the terminal log as a warning and 
WARNING: You are currently using Node.js v22.11.0, which is not supported by Hardhat. This can lead to unexpected behavior. See https://hardhat.org/nodejs-versions
Error in plugin @matterlabs/hardhat-zksync-verify: The address provided as argument contains a contract, but its bytecode doesn't match any of your local contracts.

      Possible causes are:
        - Contract code changed after the deployment was executed. This includes code for seemingly unrelated contracts.
        - A solidity file was added, moved, deleted or renamed after the deployment was executed. This includes files for seemingly unrelated contracts.
        - Solidity compiler settings were modified after the deployment was executed (like the optimizer, target EVM, etc.).
        - The given address is wrong.
        - The selected network is wrong.
ZkSyncVerifyPluginError: The address provided as argument contains a contract, but its bytecode doesn't match any of your local contracts.

      Possible causes are:
        - Contract code changed after the deployment was executed. This includes code for seemingly unrelated contracts.
        - A solidity file was added, moved, deleted or renamed after the deployment was executed. This includes files for seemingly unrelated contracts.
        - Solidity compiler settings were modified after the deployment was executed (like the optimizer, target EVM, etc.).
        - The given address is wrong.
        - The selected network is wrong.
    at inferContractArtifacts (/opt/app/api/hardhat_env/node_modules/@matterlabs/hardhat-zksync-verify/src/plugin.ts:48:45)
    at async OverriddenTaskDefinition.getContractInfo [as _action] (/opt/app/api/hardhat_env/node_modules/@matterlabs/hardhat-zksync-verify/src/task-actions.ts:310:31)
    at async Environment._runTaskDefinition (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:359:14)
    at async Environment.run (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:192:14)
    at async OverriddenTaskDefinition.verifyContract [as _action] (/opt/app/api/hardhat_env/node_modules/@matterlabs/hardhat-zksync-verify/src/task-actions.ts:197:54)
    at async Environment._runTaskDefinition (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:359:14)
    at async Environment.run (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:192:14)
    at async OverriddenTaskDefinition.verify [as _action] (/opt/app/api/hardhat_env/node_modules/@matterlabs/hardhat-zksync-verify/src/task-actions.ts:83:5)
    at async Environment._runTaskDefinition (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:359:14)
    at async Environment.run (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:192:14)
on the terminal pleas help me to solve this . thank you for your cooperation

**Comment by EngrPips** [comment]
Hello @seni7, Did you put your file in the `contracts` folder, and are you working in the `default-workspace` of Remix? Also, clear your browser cache before retrying, as that always seems to help.

**Comment by KMean** [comment]
Hey @seni7 is it working without verifying it? try what @EngrPips said and if it doesnt work try to untick verify contract on remix before deploying it. You may want to verify it manually.

**Reply by 0xMoZi** [reply]
in my case, i can do deploy the contract to zksync sepolia testnet and also interact with the contract but, i cant verifying the contract like @PatrickAlphaC do in the video or even manually, so i asked to cyfrin dc and they said isnt a big deal so i just move on learning but, still i wanna know why this happen.

**Reply by 0xMoZi** [reply]
LOL i just fixed my problem
if ur contract folder name was contract change it into contracts and change solidity version to 0.8.24 
and verify the contract directly in the remix like in the video.
idk why it worked, after 2 days concern bout this LOL

**Reply by 0xMoZi** [reply]
now my contract verifyed

**Reply by rohansai2001** [reply]
@KMean @EngrPips @fairuzStill-Learning  
Idk why mine isn't working at all @PatrickAlphaC ...I tried clearing cache and the folder structure is also correct...Could anyone suggest some solution
![image](https://github.com/user-attachments/assets/713aa9c6-8087-4bd5-9f23-5cdfc30867ef)

**Reply by EngrPips** [reply]
Please move on If you do everything correctly and the `ZKSync` doesn't just work for you. Everything as explained with Sepolia is sufficient to move on.

**Comment by ChiuMungZitAlexander** [comment]
Same here. But after I changed it to `pragma solidity 0.8.24;`, it worked.

**Reply by EngrPips** [reply]
Awesome! Thanks for the heads-up

**Comment by sklnhunt** [comment]
Adding the file to the `contracts` folder and changing the Solidity version to `0.8.24` worked for me.

**Reply by EngrPips** [reply]
Awesome! Thanks for the heads-up

**Reply by H-code9** [reply]
@EngrPips I have added the file to the contracts folder, cleared my cache, but it's only deploying and not veryfying, Please I'm stalked here, what can i do now?

**Reply by EngrPips** [reply]
Hello @H-code9, You can refresh your browser and retry, but if the issue persists, please continue with the course and don't bother with the verification.

**Reply by ZSYcho** [reply]
I have the same error. Thanks for your advice.👍

**Comment by dumbbutt0** [comment]
Hey guys so I ran into the same issue and just wanted to confirm that the fix is that the ZKSync plugin only works with solidity version {0.8.24}. i also tested with versions 23 and 25 just to be sure
<img width="400" height="900" alt="Screenshot 2025-07-20 at 9 08 40 PM" src="https://github.com/user-attachments/assets/52eb756f-7e01-432c-9173-79f865ce93a1" />
<img width="400" height="900" alt="Screenshot 2025-07-20 at 9 11 26 PM" src="https://github.com/user-attachments/assets/9b1b2bfa-bc97-4420-9d0f-5d271de877fa" />

**Reply by EngrPips** [reply, marked as answer]
Yeah, `0.8.24` seems to always work fine. I think `0.8.25` works as well.
